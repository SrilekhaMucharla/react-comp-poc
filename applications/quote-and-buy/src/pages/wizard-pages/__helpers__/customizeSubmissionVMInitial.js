const submission = {
    baseData: {
    }
};

export default submission;
