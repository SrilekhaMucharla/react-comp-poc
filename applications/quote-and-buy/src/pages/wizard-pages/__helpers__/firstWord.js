const firstWord = (sentence) => {
    let first = null;
    if (typeof sentence === 'string') {
        const words = sentence.split(' ');
        if (words.length > 0) first = words['0'];
    }
    return first;
};

export default (firstWord);
