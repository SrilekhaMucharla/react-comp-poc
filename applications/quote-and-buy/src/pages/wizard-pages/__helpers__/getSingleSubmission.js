const getSingleSubmissionVM = (submission, currentPageIndex) => submission.quotes.children[currentPageIndex] || {};
export default (getSingleSubmissionVM);
