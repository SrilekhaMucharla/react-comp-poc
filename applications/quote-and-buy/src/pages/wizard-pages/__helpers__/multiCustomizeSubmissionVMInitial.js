const multiSubmission = {
    baseData: {
    }
};

export default multiSubmission;
