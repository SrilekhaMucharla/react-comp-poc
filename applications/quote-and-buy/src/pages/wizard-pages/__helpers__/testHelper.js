const defaultTranslator = (messageID = '') => {
    let wrappedKey = messageID;
    if (typeof messageID !== 'object') {
        return messageID;
    }
    if (typeof messageID.id === 'object') {
        wrappedKey = messageID.id;
    }
    const { id = null, defaultMessage = 'defaultTranslator mock: missing both message and default message' } = wrappedKey;
    return id || defaultMessage;
};

export default defaultTranslator;
