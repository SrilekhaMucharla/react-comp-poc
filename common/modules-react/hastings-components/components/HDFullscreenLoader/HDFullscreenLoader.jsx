import React from 'react';
import PropTypes from 'prop-types';
import './HDFullscreenLoader.scss';

const HDFullscreenLoader = ({
    spinner, text, overlay
}) => {
    return (
        <>
            {overlay}
            <div className="hd-loader-container">
                {spinner}
                <div className="hd-loader-text-wrapper">
                    <span data-testid="loader-text">{text}</span>
                </div>
            </div>
        </>
    );
};

HDFullscreenLoader.propTypes = {
    spinner: PropTypes.element.isRequired,
    overlay: PropTypes.element.isRequired,
    text: PropTypes.string.isRequired,
};

export default HDFullscreenLoader;
