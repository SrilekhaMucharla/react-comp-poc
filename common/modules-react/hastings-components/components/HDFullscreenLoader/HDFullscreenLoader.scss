.hd-loader-container {
    position: fixed;
    width: 100%;
    height: 100%;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    display: flex;
    justify-content: center;
    align-items: center;
    z-index: 1000005;
    animation: 1s fadeIn ease-out;

    .hd-loader-text-wrapper {
        color: white;
        max-width: 152px;
        font-size: 24px;
        font-weight: bold;
        font-stretch: normal;
        font-style: normal;
        line-height: 0.96;
        letter-spacing: -0.48px;
        text-align: center;
        color: #ffffff;
    }
}

@keyframes fadeIn {
    from { opacity: 0; }
    to   { opacity: 1; }
}
