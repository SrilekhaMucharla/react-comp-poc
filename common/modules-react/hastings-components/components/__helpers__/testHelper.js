export const defaultTranslator = (messageID = '') => {
    const { defaultMessage = messageID } = messageID;
    return defaultMessage;
};

export const findByTestAttribute = (wrapper, selector) => wrapper.find(`[data-testid='${selector}']`);
