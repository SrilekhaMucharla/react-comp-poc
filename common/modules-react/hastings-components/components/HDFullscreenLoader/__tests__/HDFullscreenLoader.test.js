import React from 'react';
import HDFullscreenLoader from '../HDFullscreenLoader';
import { findByTestAttribute } from '../../__helpers__/testHelper';

const setup = (props) => {
    return mount(<HDFullscreenLoader {...props} />);
};

describe('<HDFullscreenLoader />', () => {
    it('correctly renders component tree', () => {
        const tree = setup({
            text: 'Test text',
            spinner: <span />,
            overlayOpacity: 0.5,
            overlayColor: 'red'
        });

        expect(tree).toMatchSnapshot();
    });

    it('should correctly display passed text', () => {
        const textToCompare = 'Loading';
        const wrapper = setup({
            text: textToCompare,
            spinner: <span />,
            overlayOpacity: 0.5,
            overlayColor: 'red'
        });
        const loaderText = findByTestAttribute(wrapper, 'loader-text').text();
        expect(loaderText).toBe(textToCompare);
    });
});
