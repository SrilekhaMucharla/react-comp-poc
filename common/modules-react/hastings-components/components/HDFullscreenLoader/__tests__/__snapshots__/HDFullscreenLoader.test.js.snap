// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HDFullscreenLoader /> correctly renders component tree 1`] = `
<HDFullscreenLoader
  overlayColor="red"
  overlayOpacity={0.5}
  spinner={<span />}
  text="Test text"
>
  <div
    className="hd-loader-container"
  >
    <span />
    <div
      className="hd-loader-text-wrapper"
    >
      <span
        data-testid="loader-text"
      >
        Test text
      </span>
    </div>
  </div>
</HDFullscreenLoader>
`;
